docs/how_to/query.md

```diff
@@ -51,6 +51,38 @@ You will notice that in the setup linked above, M3DB has just one unaggregated n
     all: false
 ```
 
+## ID generation
+
+The default generation scheme for IDs is unfortunately prone to collisions, but is left as is for back-compat reasons. It is suggested to set the ID generation scheme to one of either "quoted" or "prepend_meta". Quoted generation scheme yields the most human-readable IDs, whereas Prepend_Meta is better for more compact IDS, or if tags are expected to contain non-ASCII characters. To set the ID generation scheme, add the following to your coordinator configuration yaml file:
+
+```yaml
+tagOptions:
+  idScheme: <name>
+```
+
+As an example of how these schemes generate tags, consider a series with the following 4 tags,
+`[{"t1":v1}, {t2:"v2"}, {t3:v3}, {t4:v4}]`. The following is an example of how different schemes will generate IDs.
+
+```
+legacy: "t1"=v1,t2="v2",t3=v3,t4=v4,
+prepend_meta: 4,2,2,4,2,2,2,2!"t1"v1t2"v2"t3v3t4v4
+quoted: {\"t1\"="v1",t2="\"v2\"",t3="v3",t4="v4"}
+```
+
+If there is a chance that your metric tags will contain "control" characters, specifically `,` and `=`, it is highly recommended that one of either the `quoted` or `prepend_meta` schemes are specified, as the `legacy` scheme will be prone to ID collisions. As a general guideline, we suggest `quoted`, as it mirrors the more familiar Prometheus style IDs.
+
+We technically have a fourth ID generation scheme that is used for Graphite IDs, but it is exclusive to the Graphite ingestion path and is not selectable as a general scheme.
+
+**WARNING:** Once a scheme is selected and used, be very careful about switching it, as any incoming metrics will get an ID matching the new scheme instead, effectively doubling the metric count until any of the older metric IDs rotate out of their retention period.
+
+### Migration
+
+We have recently updated our ID generation scheme in coordinator, as we wanted to avoid the collision issues as mentioned above. Unfortunately, that can cause issues along the upgrade path, and to account for this, we're enforcing an explicitly required ID scheme generation option to be added to coordinator configuration files.
```

Comment:
coordinator -> M3Coordinator or `m3coordinator`

"We recently updated our ID generation schema in M3Coordinator to avoid the collision issues discussed above. To ease migration, we're temporarily enforcing that an ID generate scheme be explicitly provided in the M3Coordinator configuration files.

Reply:
Also maybe add, "In a future release we will lift this restriction and make the `quoted` scheme the default if none is provided in the YAML configuration"